# Google App Store EDA

# Assignment:
1. Download the Data
2. Extract data from zip file
3. read the data into python pandas
4. check the each column
5. Define each column in MD style

In [5]:
#  Import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [6]:
# Data import
df = pd.read_csv(r'C:\Users\Tariq\Desktop\vscode_environmnets\Data-Science-AI-with-Codanics\data_sets\googleplaystore.csv')

In [4]:
#  Run automatic EDA using ydata_profiling library

# comment out these codes if you want to do auto EDA
# import ydata_profiling as yd
# profile = yd.ProfileReport(df)
# profile.to_file(output_file='./outputs/ydata_googleplaystore_kaggle.html')

In [7]:
# check the data
df.info() # check the data types and missing values

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10840 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  int64  
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10841 non-null  object 
 9   Genres          10840 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10839 non-null  object 
dtypes: float64(1), int64(1), object(11)
memory usage: 1.1+ MB


In [9]:
# lets have a look at the data
df.head() # top 5 rows

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

   Reviews  Size     Installs  Type Price Content Rating  \
0      159   19M      10,000+  Free     0       Everyone   
1      967   14M     500,000+  Free     0       Everyone   
2    87510  8.7M   5,000,000+  Free     0       Everyone   
3   215644   25M  50,000,000+  Free     0           Teen   
4      967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [10]:
# take samples from big dataset
df.sample(10) # random samples from big dataset with 10 rows

App            Category  \
3444         Simple Neon Blue Future Tech Keyboard Theme     PERSONALIZATION   
4343                             Lie Detector Test Prank              FAMILY   
3591                                       My baby Phone           PARENTING   
8587                                           DN Reader  NEWS_AND_MAGAZINES   
8683                       Profile pictures for WhatsApp     PERSONALIZATION   
7874                        CT Head Basic Interpretation              FAMILY   
4966                             Ad-free Triangle Solver              FAMILY   
1955                                  BEYBLADE BURST app                GAME   
454    myMail – Email for Hotmail, Gmail and Outlook ...       COMMUNICATION   
10506                            Hair saloon - Spa salon              FAMILY   

       Rating  Reviews                Size     Installs  Type Price  \
3444      4.1    51145                6.2M  10,000,000+  Free     0   
4343      4.4   222308                6.9M  10,000,000+  Free     0   
3591      3.7     9293                2.4M   1,000,000+  Free     0   
8587      4.6       18                1.8M         100+  Free     0   
8683      4.7    10786                9.4M     500,000+  Free     0   
7874      NaN        1                 16M         100+  Free     0   
4966      4.5      118                 24M      10,000+  Free     0   
1955      4.5   216849  Varies with device  10,000,000+  Free     0   
454       4.5   305218  Varies with device  10,000,000+  Free     0   
10506     4.2    38473                 23M  10,000,000+  Free     0   

      Content Rating                     Genres     Last Updated  \
3444        Everyone            Personalization    June 28, 2018   
4343        Everyone              Entertainment    July 31, 2018   
3591        Everyone                  Parenting     May 28, 2018   
8587        Everyone           News & Magazines   March 13, 2014   
8683      Mature 17+            Personalization     May 20, 2018   
7874        Everyone                  Education    June 11, 2018   
4966        Everyone                  Education  August 20, 2015   
1955        Everyone  Action;Action & Adventure    July 16, 2018   
454         Everyone              Communication   August 3, 2018   
10506       Everyone        Casual;Pretend Play   April 11, 2018   

              Current Ver         Android Ver  
3444                    1        4.0.3 and up  
4343                3.0.2          4.4 and up  
3591            2.03.2714          4.0 and up  
8587                    1          4.0 and up  
8683                2.3.1          4.3 and up  
7874                1.0.1          4.0 and up  
4966                6.3.1          4.2 and up  
1955   Varies with device  Varies with device  
454    Varies with device  Varies with device  
10506               1.0.5          4.1 and up

In [11]:
df.describe() # statistical summary of numerical columns in the dataset

Rating       Reviews
count  9367.000000  1.084100e+04
mean      4.191513  4.441119e+05
std       0.515735  2.927629e+06
min       1.000000  0.000000e+00
25%       4.000000  3.800000e+01
50%       4.300000  2.094000e+03
75%       4.500000  5.476800e+04
max       5.000000  7.815831e+07

# Assignment: write down all the numberic variable and make them numeric.

1. size
2. install
3. price

In [16]:
#  lets have look on unique values in the dataset
df['Size'].unique() # unique values in the dataset column Size

array(['19M', '14M', '8.7M', '25M', '2.8M', '5.6M', '29M', '33M', '3.1M',
       '28M', '12M', '20M', '21M', '37M', '2.7M', '5.5M', '17M', '39M',
       '31M', '4.2M', '7.0M', '23M', '6.0M', '6.1M', '4.6M', '9.2M',
       '5.2M', '11M', '24M', 'Varies with device', '9.4M', '15M', '10M',
       '1.2M', '26M', '8.0M', '7.9M', '56M', '57M', '35M', '54M', '201k',
       '3.6M', '5.7M', '8.6M', '2.4M', '27M', '2.5M', '16M', '3.4M',
       '8.9M', '3.9M', '2.9M', '38M', '32M', '5.4M', '18M', '1.1M',
       '2.2M', '4.5M', '9.8M', '52M', '9.0M', '6.7M', '30M', '2.6M',
       '7.1M', '3.7M', '22M', '7.4M', '6.4M', '3.2M', '8.2M', '9.9M',
       '4.9M', '9.5M', '5.0M', '5.9M', '13M', '73M', '6.8M', '3.5M',
       '4.0M', '2.3M', '7.2M', '2.1M', '42M', '7.3M', '9.1M', '55M',
       '23k', '6.5M', '1.5M', '7.5M', '51M', '41M', '48M', '8.5M', '46M',
       '8.3M', '4.3M', '4.7M', '3.3M', '40M', '7.8M', '8.8M', '6.6M',
       '5.1M', '61M', '66M', '79k', '8.4M', '118k', '44M', '695k', '1.6M',
     

In [17]:
df['Size'].value_counts() # count of unique values in the dataset column Size 

Size
Varies with device    1695
11M                    198
12M                    196
14M                    194
13M                    191
                      ... 
430k                     1
429k                     1
200k                     1
460k                     1
619k                     1
Name: count, Length: 461, dtype: int64

# check that how many MBs are there in one kB? and how to calculate?
1. Convert kBs into MBs
2. Remove M from all numbers
3. Handle`varies with device`

In [18]:
df['Size'].isnull().sum() # count of missing values in the dataset column Size

0

In [25]:
# Convert size column to float by converting kB to MB and removing "kb" and "M" signs
def convert_size(size):
    if isinstance(size, str):
        if size.endswith('k'):
            return float(size[:-1]) / 1024
        elif size.endswith('M'):
            return float(size[:-1])
    return size

df['Size'] = df['Size'].replace('Varies with device', np.nan)  # Replace 'Varies with device' with NaN
df['Size'] = df['Size'].apply(convert_size)



In [28]:
df['Size'].value_counts() # count of unique values in the dataset column Size 


Size
11.000000    198
12.000000    196
14.000000    194
13.000000    191
15.000000    184
            ... 
0.419922       1
0.418945       1
0.195312       1
0.449219       1
0.604492       1
Name: count, Length: 459, dtype: int64

## for install column

In [19]:
df['Installs'].value_counts() # count of unique values in the dataset column Installs


Installs
1,000,000+        1579
10,000,000+       1252
100,000+          1169
10,000+           1054
1,000+             908
5,000,000+         752
100+               719
500,000+           539
50,000+            479
5,000+             477
100,000,000+       409
10+                386
500+               330
50,000,000+        289
50+                205
5+                  82
500,000,000+        72
1+                  67
1,000,000,000+      58
0+                  14
0                    1
Name: count, dtype: int64

# Do it as numberic column or group that into different group following bining method 

In [29]:
# - The code above removes commas and plus signs from the values and converts the column to the integer data type.

df['Installs'] = df['Installs'].str.replace(',', '').str.replace('+', '').astype(int)

In [30]:
df['Installs'].value_counts()


Installs
1000000       1579
10000000      1252
100000        1169
10000         1054
1000           908
5000000        752
100            719
500000         539
50000          479
5000           477
100000000      409
10             386
500            330
50000000       289
50             205
5               82
500000000       72
1               67
1000000000      58
0               15
Name: count, dtype: int64

# How to handle with Price?

In [20]:
df['Price'].value_counts() # count of unique values in the dataset column Price

Price
0         10041
$0.99       148
$2.99       129
$1.99        73
$4.99        72
          ...  
$19.90        1
$1.75         1
$14.00        1
$4.85         1
$1.04         1
Name: count, Length: 92, dtype: int64

In [21]:
df['Price'].isnull().sum() # count of missing values in the dataset column Price


0

## Remove $ sign from all rows using pandas

In [22]:
df['Type'].value_counts() # count of unique values in the dataset column Type


Type
Free    10040
Paid      800
Name: count, dtype: int64

#### Here in price column 10041 apps are free but in type column free apps are 10040, so one value is missing.

In [33]:
#  remove $ sign from the values and convert the column to the float data type
df['Price'] = df['Price'].str.replace('$', '').astype(float) 


In [36]:
df['Price'].value_counts()


Price
0.00     10041
0.99       148
2.99       129
1.99        73
4.99        72
         ...  
19.90        1
1.75         1
14.00        1
4.85         1
1.04         1
Name: count, Length: 92, dtype: int64

In [37]:
df.head(2)

App        Category  Rating  \
0  Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                             Coloring book moana  ART_AND_DESIGN     3.9   

   Reviews  Size  Installs  Type  Price Content Rating  \
0      159  19.0     10000  Free    0.0       Everyone   
1      967  14.0    500000  Free    0.0       Everyone   

                      Genres      Last Updated Current Ver   Android Ver  
0               Art & Design   January 7, 2018       1.0.0  4.0.3 and up  
1  Art & Design;Pretend Play  January 15, 2018       2.0.0  4.0.3 and up